src/ameedf/jb8223/assignments/hw02/Clock.java

```diff
@@ -34,4 +34,127 @@
+	public void addHours(int hours) {
+		if (hours >= 0 && hours <= 1000000000) {
+			this.hours = (byte) ((this.hours + hours) % 24);
+
+		}
+	}
+	// Validation and minutes calculation
+	public void addMinutes(int minutes) {
+		if (minutes >= 0 && minutes <= 1000000000) {
+
+			this.hours = (byte) (this.hours + (this.minutes + minutes) / 60);
+			if (this.hours > 23) {
+				this.hours = (byte) (this.hours % 24);
+			}
+
+			this.minutes = (byte) ((this.minutes + minutes) % 60);
+
+		}
+
+	}
+	// Validation and seconds calculation	
+	public void addSeconds(int seconds) {
+		if (seconds >= 0 && seconds <= 1000000000) {
+			
+			this.hours = (byte) (this.hours + (this.seconds + seconds) / 3600);
+			
+			if (this.hours > 23){
+				this.hours = (byte) (this.hours % 24);
+			}
+			
+			this.minutes = (byte) (this.minutes + (this.seconds + seconds) / 60);
+			if (this.minutes > 59){
+				this.minutes = (byte) (this.minutes % 60);
+				
+			}
+			
+			this.seconds = (byte) ((this.seconds + seconds) %  60);
+		}
+	}
+	// Validation and milliseconds calculation
+	public void addMillisecond(int milliseconds){
+		if (milliseconds >= 0 && milliseconds <= 1000000000){
+			this.hours = (byte) (this.hours + (this.milliseconds + milliseconds) / 3600000);
+			if (this.hours > 23){
+				this.hours = (byte)(this.hours %24);
+			}
+			
+			this.minutes = (byte) (this.minutes + (this.milliseconds + milliseconds) / 60000);
+			if (this.minutes > 59){
+				this.minutes = (byte)(this.minutes % 60);
+			}
+			
+			this.seconds = (byte) (this.seconds + (this.milliseconds + milliseconds) / 1000);
+			if (this.seconds > 59){
+				this.seconds = (byte) (this.seconds % 60);
+			}
+			
+			this.milliseconds = this.milliseconds % 1000;
+		
+		}	
+		
+	}
+	// 24/12 hours check + print out + formating with leading zeros
+	public void print(boolean twentyFourDisplay) {
+		if (twentyFourDisplay == false){
+			if (this.hours > 12){
+				this.hours = (byte) (this.hours % 12);
+				System.out.print("The time is ");
+				System.out.format("%02d", this.hours);
+				System.out.print(":");
+				System.out.format("%02d", this.minutes);
+				System.out.print(":");
+				System.out.format("%02d", this.seconds);
+				System.out.print(".");
+				System.out.format("%03d", this.milliseconds);
+				System.out.print(" PM");
+			}else {
+				System.out.print("The time is ");
+				System.out.format("%02d", this.hours);
+				System.out.print(":");
+				System.out.format("%02d", this.minutes);
+				System.out.print(":");
+				System.out.format("%02d", this.seconds);
+				System.out.print(".");
+				System.out.format("%03d", this.milliseconds);
+				System.out.print(" AM");
+			}
+		}else {
+			System.out.print("The time is ");
+			System.out.format("%02d", this.hours);
+			System.out.print(":");
+			System.out.format("%02d", this.minutes);
+			System.out.print(":");
+			System.out.format("%02d", this.seconds);
+			System.out.print(".");
+			System.out.format("%03d", this.milliseconds);	
+		}
```

Comment:
It's very good that you've learned the syntax of the "format" method.. BUT, think of a way to to reduce the code.
HINT: System.out.format("Your age is **%02d** and your name is **%s**", **28, "Khalil"**)